File: 기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds)
{
    vector<int> release;
    vector<int> answer;

    for (int i = 0; i < progresses.size(); i++)
    {
        int temp;
        temp = (100 - progresses[i]) % speeds[i];
        if (temp > 0)
            release.push_back((100 - progresses[i]) / speeds[i] + 1);
        else
            release.push_back((100 - progresses[i]) / speeds[i]);
    }

    int count = 1;
    int temp = release.front();

    for (int i = 0; i < release.size() - 1; i++)
    {
        if (temp >= release[i + 1])
            count++;
        else
        {
            temp = release[i + 1];
            answer.push_back(count);
            count = 1;
        }
    }

    answer.push_back(count);
    return answer;
}

File: 기능개발.cs
public class Example
{
    public static void Main(string[] args)
    {
        
    }

    public static int[] Solution(int[] progresses, int[] speeds)
    {
        Queue<int> completeQ = new Queue<int>();
        for (int i = 0; i < progresses.Length; i++)
        {
            int remain = (100 - progresses[i]) % speeds[i];
            if (remain > 0)
            {
                completeQ.Enqueue((100 - progresses[i]) / speeds[i] + 1);
            }
            else
            {
                completeQ.Enqueue((100 - progresses[i]) / speeds[i]);
            }
        }

        List<int> answers = new List<int>();
        int releaseDay = completeQ.First();
        int releaseNum = 0;
        while (completeQ.Count > 0)
        {
            if (releaseDay >= completeQ.First())
            {
                releaseNum++;
                completeQ.Dequeue();
            }
            else
            {
                answers.Add(releaseNum);
                releaseNum = 0;
                releaseDay = completeQ.First();
            }
        }
        answers.Add(releaseNum);
        return answers.ToArray();
    }
}